src/frontend/magnify/src/components/rooms/RoomConfig/RoomConfig.tsx

```diff
@@ -77,21 +72,54 @@ const messages = defineMessages({
   },
 });
 
-const RoomConfig = ({ roomName }: { roomName: string }) => {
-  const intl = useIntl();
-  const size = useContext(ResponsiveContext);
-  const initialValues = {
-    askForAuthentication: true,
-    askForPassword: true,
-    chatEnabled: true,
-    everyoneStartsMuted: true,
-    everyoneStartsWithoutCamera: true,
-    screenSharingEnabled: true,
-    waitingRoomEnabled: true,
-  };
+export interface RoomConfigProps {
+  room: Room | null | undefined;
+}
 
+const RoomConfig = ({ room }: RoomConfigProps) => {
+  const intl = useIntl();
+  const queryClient = useQueryClient();
   const isMobile = useIsMobile();
 
+  const { mutate, isLoading } = useMutation(
```

Comment:
Regarding the architecture you use to handle mutations & react query, I think it may be a good idea for you to read how we used them in richie ( search for `useCourses` `useCreditCards` etc )